modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/exception/mapper/PrincipalExceptionMapper.java

```diff
@@ -45,8 +49,82 @@ public Response toResponse(PrincipalException principalException) {
 
 	@Override
 	protected Problem getProblem(PrincipalException principalException) {
-		return new Problem(
-			Response.Status.FORBIDDEN, principalException.getMessage());
+		Locale locale = _httpServletRequest.getLocale();
+		String message;
+
+		if (principalException instanceof
```

Comment:
I think exposing the internal message could be considered a security vulnerability. Check with app sec.

Reply:
I think something like:
```
return new Problem(
			Response.Status.FORBIDDEN,
			LanguageUtil.get(_httpServletRequest.getLocale(), "forbidden"));
```

Would be better.